sklearn/pipeline.py

```diff
@@ -337,13 +362,21 @@ def fit_transform(self, X, y=None, **fit_params):
             Transformed samples
         """
         last_step = self._final_estimator
-        Xt, fit_params = self._fit(X, y, **fit_params)
-        if hasattr(last_step, 'fit_transform'):
-            return last_step.fit_transform(Xt, y, **fit_params)
-        elif last_step is None:
+        Xt, fit_params, time_elapsed_so_far = self._fit(X, y, **fit_params)
+        final_step_start_time = time.time()
+        if last_step is None:
+            if self.verbose:
+                print('[Pipeline] Step %s is NoneType.' % self.steps[-1][0])
+                print('[Pipeline] Total time elapsed: %.3fs' %
+                      time_elapsed_so_far)
             return Xt
+        elif hasattr(last_step, 'fit_transform'):
+            Xt = last_step.fit_transform(Xt, y, **fit_params)
         else:
-            return last_step.fit(Xt, y, **fit_params).transform(Xt)
+            Xt = last_step.fit(Xt, y, **fit_params).transform(Xt)
+        if self.verbose:
+            self._print_final_step(final_step_start_time, time_elapsed_so_far)
```

Comment:
I would have included `if self.verbose` condition in the method itself, to avoid repeated checks at every calls, but this way it felt semantically more appropriate - "if verbose then print final step".

Else simply the call would state "print final step" while it internally wouldn't have printed if not verbose. I hope that is fine ?